compiler/rustc_codegen_llvm/src/declare.rs

```diff
@@ -21,19 +21,43 @@ use tracing::debug;
 
 use crate::abi::{FnAbi, FnAbiLlvmExt};
 use crate::common::AsCCharPtr;
-use crate::context::CodegenCx;
+use crate::context::{CodegenCx, SimpleCx};
 use crate::llvm::AttributePlace::Function;
 use crate::llvm::Visibility;
 use crate::type_::Type;
 use crate::value::Value;
 use crate::{attributes, llvm};
 
+/// Declare a function with a SimpleCx.
+///
+/// If there’s a value with the same name already declared, the function will
+/// update the declaration and return existing Value instead.
+pub(crate) fn declare_simple_fn<'ll>(
```

Comment:
Call this from `declare_raw_fn` to deduplicate the initial logic

Reply:
done